source/game/scene/GameScene.cc

```diff
@@ -11,10 +11,37 @@ namespace Scene {
 GameScene::GameScene() {
     m_heap->setName("DefaultGameSceneHeap");
     m_nextSceneId = -1;
+
+    m_totalMemoryUsed = 0;
+    EGG::ExpHeap *heap = EGG::Heap::dynamicCastToExp(m_heap);
+    ASSERT(heap);
+
+    heap->calcGroupSize(&m_groupSizeRecord);
+    for (u16 groupID = 0; groupID < m_groupSizeRecord.size(); ++groupID) {
+        m_totalMemoryUsed += m_groupSizeRecord.getGroupSize(groupID);
+    }
 }
 
 /// @addr{0x8051A3C0}
-GameScene::~GameScene() = default;
+GameScene::~GameScene() {
+    m_resources.clear();
+
+    EGG::ExpHeap *heap = EGG::Heap::dynamicCastToExp(m_heap);
+    ASSERT(heap);
+
+    heap->calcGroupSize(&m_groupSizeRecord);
+    size_t sum = 0;
+    for (u16 groupID = 0; groupID < m_groupSizeRecord.size(); ++groupID) {
```

Comment:
Implicit cast from `size_t` to `u16`. Not sure if you wanna do anything about that, nbd either way honestly

Reply:
The comparison will upcast u16 to size_t, not downcast the size_t to u16. This is fine.